### ISSUE TITLE
Question: Response headers needs to be set/assigned again

### ISSUE THREAD

**Issue by naryal2580** [tiangolo/fastapi#3210]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [ ] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

So, the following code is simple. Sets headers and done! BUT it's not how it looks like.

```Python
from fastapi import FastAPI, Response
from fastapi.responses import RedirectResponse, FileResponse
from os.path import isfile

app = FastAPI()


@app.get("/")
def read_root(resp: Response):
    resp.headers['foo'] = 'bar'
    return RedirectResponse('https://printer.discord.com')

# OR the FileResponse
@app.get("/baz/{location:path}")
def read_baz(resp: Response):
    if isfile(location):
        return FileResponse(location)
```

### Description

* Open the browser and call the endpoint `/`.
* It returns a 307 Temporary Redirect to bla bla.. BUT WITHOUT THE **HEADERS**
* Well, the fix is simple. `RedirectResponse(..., headers=resp.headers)` BUT it actually took me a while to think of it and it would be awesome to see objects within `fastapi.responses` actually respect `fastapi.Response`.

### Environment

* OS: macOS
* FastAPI Version: 0.65.0
* Python version: Python 3.9.5

**Comment by FalseDev** [comment]
You're mutating the headers in a different object **and not returning that object**.
I'd (and probably most others) prefer this behavior over what you're suggesting.

**Comment by naryal2580** [comment]
Exactly, but would cool to see objects within `fastapi.responses` to enumerate values from `fastapi.Response`

Let's keep the issue open, and wait for some other views.

**Comment by Kludex** [comment]
The thing is that if you explicitly return a Response (any kind), it will overwrite your response object. This is seen as a feature, not an issue.

**Comment by naryal2580** [comment]
Makes sense, another object is another object, and objects acting polymorphic could be a nightmare while debugging.

**Comment by tiangolo** [comment]
Thanks for the help here @FalseDev and @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @naryal2580  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
